dialog/file.go

```diff
@@ -547,26 +554,25 @@ func (f *fileDialog) setView(view ViewLayout) {
 		if dir, ok := f.getDataItem(id); ok {
 			parent := id == 0 && len(dir.Path()) < len(f.dir.Path())
 			_, isDir := dir.(fyne.ListableURI)
-			o.(*fileDialogItem).setLocation(dir, isDir || parent, parent)
+			o.(*fileDialogItem).setLocation(dir, isDir || parent, parent, id)
+			o.(*fileDialogItem).choose = choose
+			o.(*fileDialogItem).open = f.open.OnTapped
 		}
 	}
-	choose := func(id int) {
-		if file, ok := f.getDataItem(id); ok {
-			f.selectedID = id
-			f.setSelected(file, id)
-		}
-	}
-
+	// Acutally, during the real interaction, the choose won't be called.
```

Comment:
I'm not sure what this comment is really communicating? Also there is a typo misspelling in "Actually"

Reply:
Yes, a typo. I mean The choose function won't be called through select because we override the tapped function.